src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPECompilerManager.cs

```diff
@@ -204,7 +211,11 @@ private void QueueCompilation()
         {
             if (_filesToCompile.Count > 0)
             {
-                _scheduler.ScheduleAsyncTask(ProcessCompileQueueAsync, _project.Services.ProjectAsynchronousTasks.UnloadCancellationToken);
+                JoinableTask task = _scheduler.ScheduleAsyncTask(ProcessCompileQueueAsync, _project.Services.ProjectAsynchronousTasks.UnloadCancellationToken);
```

Comment:
I would consider setting a TaskCOmpletetionSource that you can wait on instead.

Reply:
Or just wait for the telemetry?

Reply:
The tests are using a TaskCompletionSource, but for scenarios where there is not supposed to be anything compiled it doesn't help. Making this optionally synchronous seems to have fixed the flakiness in this, its really the dataflow that is giving me trouble. I have one more thing to try for that though.